pkg/controllers/node/liveness.go

```diff
@@ -36,7 +36,10 @@ type Liveness struct {
 
 // Reconcile reconciles the node
 func (r *Liveness) Reconcile(ctx context.Context, provisioner *v1alpha3.Provisioner, n *v1.Node) (reconcile.Result, error) {
-	if !node.FailedToJoin(n, LivenessTimeout) {
+	if Now().Sub(n.GetCreationTimestamp().Time) < LivenessTimeout {
```

Comment:
This seems a little convoluted compared to the `.Since()` function in the time library. It’s essentially the same functionality, but I think Since() looks nicer.

Reply:
I'd need to mock since as well. Tough tradeoff.